### PR TITLE
Improve subsetter

### PR DESCRIPTION
**Pull request by LaurenzV**
Still in-progress and far from being done. Just opening this so people know that it is being worked on.

### CODE REVIEW

**Comment by LaurenzV** [comment]
@laurmaedje so basically, I think this is ready to be reviewed now. I won't write much more here in the comment about the implementation itself, as I've tried to add README's/comments in the code wherever possible, but let me know if something is unclear.

I've let the fuzzer run for 20 minutes now and no issues have arised, so I'm (cautiously) feeling pretty good about it.

We do need to test more fonts for PDF (right now I've just tested 3 different ones to make sure the integration works), but I would prefer to do this with the PR in Typst, to make sure that you are okay with the current design of the crate, before I invest more time in testing that. And then we can also test it with some printers if we can find volunteers. So I think this shouldn't block this PR.

Let me know what you think (but no rush, as I know this is a lot, and I also don't expect you to try to understand every detail of how it works 😄)!

**Comment by laurmaedje** [comment]
Great, I'll try to review it soon!

File: src/lib.rs [hunk 1]
```diff
@@ -168,9 +168,11 @@ fn parse(data: &[u8], index: u32) -> Result<Face<'_>> {
     let mut r = Reader::new(data);
     let mut kind = r.read::<FontKind>().ok_or(UnknownKind)?;
 
+    // TODO: Add a test font for ttc.
+
```
Comment (LaurenzV) [inline src/lib.rs:172]: Note to myself: Don't forget about this.

**Comment by LaurenzV** [comment]
So I just tried changing the skrifa fuzzer to use hinting and seems like there are often small differences... Will try to investigate this.

**Comment by LaurenzV** [comment]
Alright, I did end up finding two issues, but there are still other issues with the hinted drawing mode... However, I get the same problem when using `fonttools` subset, so I don't think this should be a blocker. I also asked about it at https://github.com/googlefonts/fontations/issues/936

**Comment by laurmaedje** [comment]
Thank you so much! 🚀🚀

**Pull request merged**